src/promptflow/tests/executor/unittests/_utils/test_tool_utils.py

```diff
@@ -360,3 +362,29 @@ def test_load_function_from_function_path_with_error(self, mock_module_with_list
             "is not callable.'. \nPlease contact the tool author/support team for troubleshooting assistance.",
         ):
             load_function_from_function_path(func_path)
+
+    @pytest.mark.parametrize(
+        "func_call_scenario, result, err_msg",
+        [
+            (
```

Comment:
Add the invalid result for dynamic list to this part so that we can directly remove the dynamic_list related code.

Reply:
updated.